### PR TITLE
Add activity bar webview in focused review mode

### PR DESCRIPTION
**Pull request by RMacfarlane**
I moved the current webview stuff around, so this looks like quite a large changeset.  I renamed 'preview-src' to 'webviews' and tried to organize the files there a bit more by categorizing things either into "components" or "common". I then added separate entry points for the current description webview and the new activity bar one there.

I copy-pasted a bunch of code from `pullRequestOverview` to the webview view provider, I want to clean this up and de-duplicate it. Will probably do that tomorrow

### CODE REVIEW

File: src/view/reviewManager.ts [hunk 1]
```diff
@@ -59,11 +60,13 @@ export class ReviewManager {
 	private _isFirstLoad = true;
 
 	constructor(
+		private _context: vscode.ExtensionContext,
 		private _repository: Repository,
 		private _folderRepoManager: FolderRepositoryManager,
 		private _telemetry: ITelemetry,
 		public changesInPrDataProvider: PullRequestChangesTreeDataProvider
 	) {
+		console.log(_context);
```
Comment (RMacfarlane) [inline src/view/reviewManager.ts:69]: remove

File: src/view/reviewManager.ts [hunk 2]
```diff
@@ -261,6 +264,9 @@ export class ReviewManager {
 		Logger.appendLine(`Review> register comments provider`);
 		await this.registerCommentController();
 
+		const webviewViewProvider = new PullRequestViewProvider(this._context.extensionUri, this._folderRepoManager, pr);
+		this._context.subscriptions.push(vscode.window.registerWebviewViewProvider(PullRequestViewProvider.viewType, webviewViewProvider));
```
Comment (RMacfarlane) [inline src/view/reviewManager.ts:268]: should actually dispose this when the active PR changes

File: src/github/activityBarViewProvider.ts [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+						if (isBranchActive) {
+							if (this._folderRepositoryManager.repository.state.workingTreeChanges.length) {
+								const response = await vscode.window.showWarningMessage(`Your local changes will be lost, do you want to continue?`, { modal: true }, 'Yes');
+								if (response === 'Yes') {
+									await vscode.commands.executeCommand('git.cleanAll');
+								} else {
+									return;
+								}
+							}
+							// await this._folderRepositoryManager.repository.checkout(this._repositoryDefaultBranch);
```
Comment (alexr00) [inline src/github/activityBarViewProvider.ts:256]: delete?

File: src/github/activityBarViewProvider.ts [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+						return await this._folderRepositoryManager.repository.deleteBranch(branchInfo!.branch, true);
+					case 'remote':
+						return this._folderRepositoryManager.repository.removeRemote(branchInfo!.remote!);
+				}
+			});
+
+			await Promise.all(promises);
+
+			// this.refreshPanel();
+			vscode.commands.executeCommand('pr.refreshList');
```
Comment (alexr00) [inline src/github/activityBarViewProvider.ts:267]: Maybe during October we do a purge of all of these `executeCommand`s

File: .vscodeignore [hunk 5]
```diff
@@ -1,7 +1,6 @@
 .vscode/**
 out/**
 src/**
-preview-src/**
```
Comment (alexr00) [inline .vscodeignore:4]: Does `webviews` need to be added here now? We need to stay under 25 files to publish for web.

File: package.json [hunk 6]
```diff
@@ -1209,7 +1215,7 @@
     "vscode:prepublish": "webpack --env.production --config-name webview && webpack --env.production --config-name extension && npx webpack-cli --config browser.webpack.config --env.production --config-name extension --mode none",
```
Comment (alexr00) [inline package.json:1215]: Any script that has `--config-name webview` needs to be updated to include the appropriate `--config-name` for the activity bar webview.

Comment (alexr00) [reply package.json:1215]: and `--config-name webview` needs to be changed to `--config-name webviewIndex.js`

**Review by alexr00 — APPROVED** [review]
Just a few comments about updating the scripts and .vscodeignore. Looks good.

File: src/github/activityBarViewProvider.ts (hunk 4 above)
Comment (RMacfarlane) [reply src/github/activityBarViewProvider.ts:267]: yes, definitely!

File: package.json (hunk 6 above)
Comment (RMacfarlane) [reply package.json:1215]: Thanks for catching this!

**Pull request merged**